src/routes/demographics/Content.svelte

```diff
@@ -143,21 +166,61 @@
           {#if schema[question].type === 'text'}
             <div
               class="mzp-c-choice mzp-c-choice-text"
-              class:mzp-is-error={!$zipValidity.valid}>
+              class:mzp-is-error={
+                inputFormatters.showErrors(question) && // only show errors if explicitly told, e.g. on blur
+                inputFormatters.hasValidator(question) && // only show errors if there's even a validator
+                inputFormatters[question].isInvalid(results[question] // only show errors if the input is invalid
+            )}>
```

Comment:
nit paired with the previous ;)
```suggestion
            }>
```